src/floyd-warshall.jl

```diff
@@ -36,17 +36,17 @@ end
 #     Note that it is possible to consume large amounts of memory as the
 #     space required for the FloydWarshallState is O(n^2).
 #     """ ->
-function floyd_warshall_shortest_paths(
+function floyd_warshall_shortest_paths{T<:Real}(
     g::AbstractGraph;
-    edge_dists::AbstractArray{Float64, 2} = Array(Float64,(0,0))
+    edge_dists::AbstractArray{T,2} = zeros(0,0)
```

Comment:
No. A method definition with a default argument is exactly equivalent to two separate method definitions, one of which sets the default value.

```jl
f{T}(x::T=0) = x
```

is exactly equivalent to

```jl
f{T}(x::T) = x
f() = f(0)
```

Therefore, the default argument must already specify a concrete type and cannot refer to a typevar. (You cannot have a method signature `f{T}()`.)

Were you to change the method definition to `f{T}(x::T=zero(T))`, you would get an error because `T` is not defined.